src/config/config_type.h

```diff
@@ -186,15 +186,17 @@ class YesNoField : public ConfigField {
 
 class EnumField : public ConfigField {
  public:
-  EnumField(int *receiver, ConfigEnum *enums, int e) : receiver_(receiver), enums_(enums) { *receiver_ = e; }
+  EnumField(int *receiver, std::vector<ConfigEnum> &enums, int e) : receiver_(receiver), enums_(std::move(enums)) {
```

Comment:
```suggestion
  EnumField(int *receiver, std::vector<ConfigEnum> enums, int e) : receiver_(receiver), enums_(std::move(enums)) {
```

Reply:
otherwise the passed `enums` will be invalid after the ctor call, which is very counterintuitive since it is passed as lvalue reference.

Reply:
Thanks @PragmaTwice, it makes a lot of sense